spec.md.html

```diff
@@ -435,19 +425,99 @@
+\end{eqnarray}
+
+We are only interested in the difference between $\phi_l$ and $\phi_v$, because rotating the BRDF does not change the result. Therefore, with $\phi=\phi_l - \phi_v$ and $\phi_v = 0$ ($\implies \mathrm d \phi = \mathrm d \phi_l$) we get
+
+\begin{eqnarray}
+E_m(\theta_v) &=& \int_{0}^{2\pi} \int_{0}^{\frac{\pi}{2}} M_r(\theta_l, \phi, \theta_o) \cos \theta_l \sin \theta_l \, \mathrm d \theta_l \, \mathrm d \phi
+\end{eqnarray}
+
+By reparameterization with $\mu_{l,v} = \cos \theta_{l,v}$ ($\implies \mathrm d \mu_{l,v} = -\sin \theta_{l,v} \, \mathrm d \theta_{l,v}$), we arrive at the simpler form
+
+\begin{eqnarray}
+E_m(\mu_v) = \int_{0}^{2\pi} \int_{0}^{1} M_r(\mu_l, \phi, \mu_v)  \, \mu_l \, \mathrm d \mu_l \, \mathrm d \phi
+\end{eqnarray}
+
+Evaluating this integral for different values of $\mu_v$ and isotropic roughness $r$ results in the image shown in Figure [energy-compensation-ggx-e].
+
+![Figure [energy-compensation-ggx-e]: Directional albedo of single-scattering microfacet GGX BRDF](img/energycomp_ggx_e.png)
+
+Integrating over the hemisphere once more gives us the average albedo (Figure [energy-compensation-ggx-e-avg]):
+
+\begin{eqnarray}
+E_{m, avg} &=& 2 \int_0^1 E_{m}(\mu_v) \mu_v \, \mathrm d \mu_v
+\end{eqnarray}
+
+![Figure [energy-compensation-ggx-e-avg]: Average albedo of single-scattering microfacet GGX BRDF](img/energycomp_ggx_e_avg.png)
+
+We are now able to store these results in a lookup table (e.g., with a size of 32x32 pixels) and fetch them during the evaluation of the BSDF to compute the energy compensation term $M_{r,ms}(\mathbf{v},\mathbf{l})$.
+
+Including the Fresnel term [#Hil18], the complete multiple-scattering GGX BRDF looks as follows:
+
+\begin{eqnarray}
+f_{r}(\mathbf{v},\mathbf{l}) &=& M_r(\mathbf{v},\mathbf{l}) \, F(\mathbf{h_r}) + M_{r,ms}(\mathbf{v},\mathbf{l}) \, F_{ms} \\
+&=& M_r(\mathbf{v},\mathbf{l}) \, F(\mathbf{h}_r \cdot \mathbf{v})  + \frac{(1 - E_m(\mathbf{v}\cdot\mathbf{n})(1 - E_m(\mathbf{l}\cdot\mathbf{n}))}{\pi (1 - E_{m,avg})} \frac{{F_{m,avg}}^2 E_{m,avg}}{1 - F_{m,avg} (1 - E_{m,avg})}
+\end{eqnarray}
+
+with the average of the Schlick Fresnel:
+
+\begin{eqnarray}
+F_{m,avg} &=& \frac{1}{21} \rho_{s,r,90} + \frac{20}{21} \rho_{s,r,0}
+\end{eqnarray}
+
+We provide the lookup tables for download in our Github repository:
+* [res/GGX_E.exr](https://github.com/DassaultSystemes-Technology/EnterprisePBRShadingModel/tree/master/res/GGX_E.exr)
+* [res/GGX_E_avg.exr](https://github.com/DassaultSystemes-Technology/EnterprisePBRShadingModel/tree/master/res/GGX_E_avg.exr)
+
+### Diffuse BRDF
+
+The light that hits the surface will either reflect from the specular component or enter the surface, bounce several times in the medium, and leave at a random surface point in a random direction. This results in a diffuse look that is modeled with the diffuse component $B(\mathbf{v}, \mathbf{l})$. The ratio between specular and diffuse reflection is determined by the Fresnel term.
```

Comment:
> 
**The light that hits the surface will either reflect from the specular component or enter the surface, bounce several times in the medium, and leave at a random surface point in a random direction.**

This sentence is very confusing here. Are we talking about fresnel weighted diffuse component under multiple scattering GGX microfacet or SSS?